packages/features/calendars/CalendarSwitch.tsx

```diff
@@ -69,15 +76,23 @@ const CalendarSwitch = (props: ICalendarSwitchProps) => {
       <div className="flex pl-2">
         <Switch
           id={externalId}
-          defaultChecked={isChecked}
+          checked={checkedInternal}
           onCheckedChange={(isOn: boolean) => {
+            setCheckedInternal(isOn);
             mutation.mutate({ isOn });
           }}
         />
       </div>
       <label className="ml-3 text-sm font-medium leading-5" htmlFor={externalId}>
         {name}
       </label>
+      {!!props.destination && (
+        <span className="bg-subtle text-default ml-8 inline-flex items-center gap-1 rounded-md px-2 py-1 text-sm font-normal sm:ml-4">
+          <ArrowLeft className="h-4 w-4" />
+          {t("adding_events_to")}
+        </span>
+      )}
+      {mutation.isLoading && <RotateCcw className="text-muted h-4 w-4 animate-spin ltr:ml-1 rtl:mr-1" />}
```

Comment:
Add a spinner to let the user know the switch is doing something - without it you can accidentally navigate away when the request is being fired.